### PR TITLE
Shipments list view

### PR DESCRIPTION
**Pull request by nvthanh11**
Create shipments list view

### CODE REVIEW

File: shuup/admin/templates/shuup/admin/base_picotable.jinja [hunk 1]
```diff
@@ -15,4 +15,31 @@
 {% endblock %}
 {% block extra_js %}
 <script src="{{ shuup_static("shuup_admin/js/picotable.js") }}"></script>
+{% if listview %}
```
Comment (chessbr) [inline shuup/admin/templates/shuup/admin/base_picotable.jinja:18]: this script shouldn't be here, but in the a custom template, like here: `shuup/simple_supplier/templates/shuup/simple_supplier/admin/base_picotable.jinja`

File: shuup/admin/templates/shuup/admin/base_picotable.jinja [hunk 2]
```diff
@@ -15,4 +15,31 @@
...
+<script>
+    function handleSetSent(event, link) {
+            event.preventDefault();
+            var warningMessage = "{% trans %}Do you want to mark this shipment as sent?{% endtrans %}";
+            if (confirm(warningMessage)) {
+                var form = document.createElement("form");
+                form.method = "POST";
+                console.log($(link).attr("href"));
```
Comment (chessbr) [inline shuup/admin/templates/shuup/admin/base_picotable.jinja:26]: remove this log

File: shuup/admin/modules/orders/views/list.py [hunk 3]
```diff
@@ -91,3 +91,161 @@ def get_object_abstract(self, instance, item):
...
+            return super(ShipmentListView, self).get_queryset().exclude(status=ShipmentStatus.DELETED)
+        return super(ShipmentListView, self).get_queryset().exclude(status=ShipmentStatus.DELETED).filter(supplier=get_supplier(self.request))
+
+    def get_object_abstract(self, instance, item):
+        return [
+            {"text": "%s" % instance, "class": "header"},
+            {"title": _(u"Content"), "text": item.get("get_content")},
+        ]
+    def post(self, request, *args, **kwargs):
+        shipment = Shipment.objects.get(pk=request.POST['shipment_id'])
```
Comment (chessbr) [inline shuup/admin/modules/orders/views/list.py:247]: there is a view to set a shipment as sent, the `ShipmentSetSentView`, use it instead

File: shuup/admin/modules/orders/views/list.py [hunk 4]
```diff
@@ -91,3 +91,161 @@ def get_object_abstract(self, instance, item):
...
+        for i in query:
+            product = product + f"{i.product} ({i.quantity})"
+        return product
+
+    def create_action_buttons(self, instance):
+        if instance.status == ShipmentStatus.SENT:
+            return instance.status.name
+        return f'<a class="btn btn-info btn-sm set_shipment_status" \
+            href="{reverse("shuup_admin:order.shipments-list")}" shipment_id="{instance.id}" \
+            onclick="handleSetSent(event, this)"><i class="fa fa-truck"></i> Mark as sent</a>'
```
Comment (chessbr) [inline shuup/admin/modules/orders/views/list.py:231]: make sure to translate the Mark as sent string

File: shuup/admin/modules/orders/views/list.py [hunk 5]
```diff
@@ -91,3 +91,161 @@ def get_object_abstract(self, instance, item):
...
+        self.columns = self.default_columns
+    def get_queryset(self):
+        if get_supplier(self.request) is None:
+            return super(ShipmentListView, self).get_queryset().exclude(status=ShipmentStatus.DELETED)
+        return super(ShipmentListView, self).get_queryset().exclude(status=ShipmentStatus.DELETED).filter(supplier=get_supplier(self.request))
+
+    def get_object_abstract(self, instance, item):
+        return [
+            {"text": "%s" % instance, "class": "header"},
+            {"title": _(u"Content"), "text": item.get("get_content")},
```
Comment (chessbr) [inline shuup/admin/modules/orders/views/list.py:244]: make sure to also show the status, the button to set as sent and other columns as well

File: shuup/admin/modules/orders/views/list.py [hunk 6]
```diff
@@ -91,3 +91,161 @@ def get_object_abstract(self, instance, item):
             {"title": _(u"Total"), "text": item.get("taxful_total_price_value")},
             {"title": _(u"Status"), "text": item.get("status")}
         ]
+# -*- coding: utf-8 -*-
```
Comment (chessbr) [inline shuup/admin/modules/orders/views/list.py:94]: what is this code going here?

File: shuup/admin/modules/orders/views/list.py [hunk 7]
```diff
@@ -91,3 +91,161 @@ def get_object_abstract(self, instance, item):
...
+    ChoicesFilter, Column, DateRangeFilter, MultiFieldTextFilter, RangeFilter,
+    TextFilter
+)
+from shuup.admin.utils.views import PicotableListView
+from shuup.core.models import Order, OrderStatus, PaymentStatus, ShippingStatus, Shipment, ShipmentStatus, ShipmentProduct
+from shuup.utils.django_compat import reverse
+from shuup.utils.i18n import format_money, get_locally_formatted_datetime
+
+
+class OrderListView(PicotableListView):
```
Comment (chessbr) [inline shuup/admin/modules/orders/views/list.py:121]: why this view here?

File: shuup/admin/modules/orders/__init__.py [hunk 8]
```diff
@@ -207,3 +207,27 @@ def get_menu_entries(self, request):
 
     def get_model_url(self, object, kind, shop=None):
         return derive_model_url(OrderStatus, "shuup_admin:order_status", object, kind)
+
+
+class ShipmentModule(AdminModule):
+    name = _("Shipments")
+    breadcrumbs_menu_entry = MenuEntry(name, url="shuup_admin:order.shipments-list")
+    def get_urls(self):
```
Comment (chessbr) [inline shuup/admin/modules/orders/__init__.py:215]: move both the URLs and the menu entries to the `OrderModule`

**Review by chessbr — CHANGES_REQUESTED** [review]

File: shuup/admin/modules/orders/views/list.py (hunk 6 above)
Comment (nvthanh11) [reply shuup/admin/modules/orders/views/list.py:94]: a mistake copy-paste

File: shuup/admin/modules/orders/views/list.py (hunk 3 above)
Comment (nvthanh11) [reply shuup/admin/modules/orders/views/list.py:247]: I made a Listview for Shipments so I want to use it instead, your view currently redirects to an Orderdetail view, and it is dealing with Post request in a different way my Listview does. So I prefer to use my Listview.

**Comment by nvthanh11** [comment]
I'll create new PR for this

**Pull request closed without merging**